modules/data_release/ajax/AddPermission.php

```diff
@@ -20,9 +20,27 @@
 $baseURL  = $settings->getBaseURL();
 $DB       = $factory->database();
 
-if (isset($_POST['action']) && $_POST['action'] == 'addpermission'
-    && $user->hasPermission('data_release_edit_file_access')
-) {
+
+if (!isset($_GET['action'])) {
+    header('');
+    exit;
+}
+
+if (!$user->hasPermission('data_release_edit_file_access')) {
+    header("HTTP/1.1 403 Forbidden");
+    exit;
+}
+
+
+if ($_GET['action'] == 'addpermission') {
+
+    if (!isset($_POST) || empty($_POST)) {
```

Comment:
It should be either a GET or a POST right? I don't think both of these arrays will have data at the same time.

Reply:
With Ajax you can send data to a PHP script via GET and POST in the same request. 

From: https://coursesweb.net/ajax/get-post

Since this is an ajax request, I guess it is fine. (it is definitely working when I test it I think)

Reply:
Hmmm okay. I don't think this is good practice but it can be handled in another request.

I believe the correct way to use GET is that it should not modify anything on the server where POST will. It's better to keep these methods separate instead of mixing them.

Reply:
That makes sense. Just funny to mix them.